### PR TITLE
Fix doc: set_callback also accepts an array of if:

### PR DESCRIPTION
**Pull request by claudiob**
When Active Record calls `set_callback` inside `after_commit`, [these lines of code](https://github.com/rails/rails/blob/master/activerecord/lib/active_record/transactions.rb#L276) pass an **array** of methods as the `:if` condition:

``` ruby
options[:if] = Array(options[:if])
options[:if] << "transaction_include_any_action?(#{fire_on})"
```

That made me realize that anyone could pass an **array** of `:if` and `:unless` conditions to `set_callback`, since Active Support transforms these conditions into an array anyways in [these lines of code](https://github.com/rails/rails/blob/master/activesupport/lib/active_support/callbacks.rb#L365):

``` ruby
@if      = Array(options[:if])
@unless  = Array(options[:unless])
```

Long story short, this commit updates the documentation of the `set_callback` method to explain that arrays are also accepted.

It also replaces `+false+` and `+true+` with `false` and `true`, since any _falsey_ or _truthy_ value will work.

[ci skip]

### CODE REVIEW

File: activesupport/lib/active_support/callbacks.rb [hunk 1]
```diff
@@ -663,10 +663,12 @@ def __update_callbacks(name) #:nodoc:
       #
       # ===== Options
       #
-      # * <tt>:if</tt> - A symbol naming an instance method or a proc; the
-      #   callback will be called only when it returns a +true+ value.
-      # * <tt>:unless</tt> - A symbol naming an instance method or a proc; the
-      #   callback will be called only when it returns a +false+ value.
+      # * <tt>:if</tt> - A symbol, a string or an array of symbols and strings,
+      #   each naming an instance method or a proc; the callback will be called
+      #   only when they all return a +true+ value.
```
Comment (senny) [inline activesupport/lib/active_support/callbacks.rb]: A think we should not quote `true` as we are referring to Ruby's definition of a "true value" and not the exact value `true`.

Comment (claudiob) [inline activesupport/lib/active_support/callbacks.rb]: I totally agree! I had the same thought when I reviewed my PR :ribbon:

**Comment by senny** [comment]
@claudiob thanks man! :yellow_heart:
